crates/solidity/inputs/schema/grammar/05-expressions/03-numbers/productions.yml

```diff
@@ -2,43 +2,78 @@
 
 - name: "NumericLiteral"
   kind: "Parser"
-  unversioned:
-    sequence:
-      - choice:
-          - reference: "HexLiteral"
-          - reference: "DecimalLiteral"
-      - optional:
-          reference: "NumberUnit"
+  versioned:
+    0.4.11:
+      sequence:
+        - choice:
+            - reference: "HexLiteral"
+            - reference: "DecimalLiteral"
+        - optional:
+            reference: "NumberUnit"
+    0.5.0:
+      sequence:
+        - reference: "DecimalLiteral"
```

Comment:
I wonder if moving from `choice` to `sequence` can change the shape of the tree?
If so, should `0.5.0` be a `choice` of a single child to maintain it?

Reply:
For others as well.

Reply:
1. No, choice doesn't add a node.
2. I don't like the idea of later versions carrying the baggage of earlier versions.